app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/ExtractOfflineProductService.java

```diff
@@ -0,0 +1,246 @@
+import android.content.SharedPreferences;
+import android.os.AsyncTask;
+import android.os.Environment;
+import android.support.annotation.Nullable;
+import android.support.v4.app.NotificationCompat;
+import android.util.Log;
+
+import com.opencsv.CSVReader;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import openfoodfacts.github.scrachx.openfood.R;
+import openfoodfacts.github.scrachx.openfood.models.OfflineProduct;
+import openfoodfacts.github.scrachx.openfood.models.OfflineProductDao;
+import openfoodfacts.github.scrachx.openfood.utils.Utils;
+import openfoodfacts.github.scrachx.openfood.views.MainActivity;
+
+
+public class ExtractOfflineProductService extends IntentService {
+
+    private static final String TAG = "EOPService";
+    public static final String IDENTIFIER_ = "INTENT_IDENTIFIER__FOR_BRAODCASTING";
+    public static final String EXTRACT_PROGRESS_UPDATE_KEY = "PROGRESS_UPDATE_FOR_EXTRACTING";
+    public static boolean isExtractOfflineProductServiceRunning = false;
+    NotificationManager notificationManager;
+    NotificationCompat.Builder builder;
+    SharedPreferences settings;
+    Intent broadcastIntent;
+
+    public ExtractOfflineProductService() {
+        super("ExtractOfflineProductService");
+        broadcastIntent = new Intent(IDENTIFIER_);
+    }
+
+    @Override
+    protected void onHandleIntent(@Nullable Intent intent) {
+        isExtractOfflineProductServiceRunning = true;
+        settings = getSharedPreferences("prefs", Context.MODE_PRIVATE);
+        notificationManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
+            String channelId = "channel";
+            CharSequence channelName = getString(R.string.notification_channel_name);
+            int importance = NotificationManager.IMPORTANCE_DEFAULT;
+            NotificationChannel notificationChannel = new NotificationChannel(channelId, channelName, importance);
+            notificationManager.createNotificationChannel(notificationChannel);
+        }
+        builder = new NotificationCompat.Builder(this, "channel");
+        builder.setContentTitle(getString(R.string.app_name))
+                .setContentText(getString(R.string.txtDownloading))
+                .setAutoCancel(true)
+                .setSmallIcon(R.mipmap.ic_launcher);
+
+        unzip(new File(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS) + File.separator + "fr.openfoodfacts.org.products.small.zip"), Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS).getAbsolutePath());
```

Comment:
Can you turn that into a non-translatable string so that it can work with Open Beauty Facts and the other variants ?